### ISSUE TITLE
CollectionView's layout breaks sometimes when a page is closed

### ISSUE THREAD

**Issue by cpetrov** [eclipsesource/tabris-js#1417]
### Problem description
CollectionView layout sometimes breaks when a page on top of its containing page in the stack is disposed.

### Expected behavior
CollectionView layout should not change when pages on top of its containing page are disposed.

### Environment

- Tabris.js version: 2.x-nightly
- Device: Nexus 5X (simulator)
- OS: 7.1.1

### Code snippet

Tap on the text container. A new page will open. Close it. The layout of the CollectionView will break (its bottom bound will be wrong).

It seems longer texts produce higher values for the bottom bound of the CollectionView when reproducing the issue. Try changing `repeat(1)` to `repeat(3)` which will cause the CollectionView to completely disappear when the page on the top is closed.

The issue cannot be reproduced when using e.g. `ScrollView` or `Composite` with children instead of the `CollectionView`.
```javascript
const {NavigationView, Composite, CollectionView, Page, TextView, ui} = require('tabris');

const COLORS = ['red', 'green', 'blue', '#00FFFF', '#FF00FF', 'yellow', 'black'];

let navigationView = new NavigationView({left: 0, top: 0, right: 0, bottom: 0})
  .appendTo(ui.contentView);

let page = new Page();

let titleComposite = new Composite({
  left: 0, right: 0, top: 0
}).on('tap', () => {
  new Page().appendTo(navigationView);
}).appendTo(page);

new TextView({
  left: 8, top: 8, right: 8,
  text: 'Lorem ipsum dolor sit amet.'.repeat(1) // try 1 and 3
}).appendTo(titleComposite);

new CollectionView({
  left: 0, top: titleComposite, right: 0, bottom: 0,
  itemCount: COLORS.length,
  cellHeight: 128,
  createCell: () => new Composite({highlightOnTouch: true}),
  updateCell: (cell, index) => cell.background = COLORS[index]
}).appendTo(page);

page.appendTo(navigationView);
```

**Comment by mpost** [comment]
When hiding the underlying page containing the `CollectionView` we set the underlying page visibility to `GONE` on Android. This leads to the confusion where the `CollectionView` is not properly layed out when the page is set to `VISIBLE` again.

The problem can be circumvented by just setting the page visibility to `INVISIBLE` but this will cause all pages underneath the top one to also be layed out.

The problem might be related to the issue https://issuetracker.google.com/issues/37138840 which is marked as fixed but i am not sure whether that is included in the constraint layout library 1.1.0-beta1 which we are currently using. 

The problem can also be replicated by just relayouting a freshly added page. We should create a self contained native snippet to open a new issue on the android issue tracker.

**Comment by mpost** [comment]
After benchmarking the layouting time with many pages we came to the conclusion that the layout time is not affected by setting the page visibility to `INVISIBLE` rather than `GONE`. Since the pages do not reference each other (they are always take the entire size of the navview), the layouting is likely smart enough to limit the relayouting to the affected pages.
